File: Data Structures & Algorithms/validate-parentheses/submission-8.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.length() % 2 != 0){
            return false;
        }

        stack<char> stk;

        for(char c:s){
            if(c == '(' || c == '{' || c == '['){
                stk.push(c);
            } else if(stk.empty() || (c == ')' && stk.top() != '(') ||
                (c == '}' && stk.top() != '{') || 
                (c == ']' && stk.top() != '[')){
                return false;
            } else {
                stk.pop();
            }
        }

        return stk.empty();
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-8.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for tok in tokens:
            if tok == "+":
                second = int(stack.pop())
                first = int(stack.pop())
                stack.append(str(first + second))
            elif tok == "-":
                second = int(stack.pop())
                first = int(stack.pop())
                stack.append(str(first - second))
            elif tok == "*":
                second = int(stack.pop())
                first = int(stack.pop())
                stack.append(str(first * second))
            elif tok == "/":
                second = int(stack.pop())
                first = int(stack.pop())
                stack.append(str(int(first / second)))
            else:
                stack.append(tok)

        return int(stack[-1])